pyfstat/snr.py

```diff
@@ -0,0 +1,436 @@
+from attrs import define, field
+import lal
+import lalpulsar
+import logging
+import numpy as np
+from typing import Union
+
+from pyfstat.helper_functions import get_ephemeris_files
+
+
+@define(kw_only=True, slots=False)
+class SignalToNoiseRatio:
+    """Compute the optimal SNR of a CW signal on Gaussian noise or an arbitrary dataset of SFTs.
```

Comment:
Actually this is a bit misleading, as it by definition always assumes Gaussian noise, and there is no non-Gaussianity information extracted from the SFTs. (At most, non-stationarity.) So let's simplify that part away from the 1-line doc.

```suggestion
    """Compute the optimal SNR of a CW signal as expected in Gaussian noise.
```

Reply:
`(non-)stationary Gaussian noise`?

Reply:
Let's keep it simple, that addition is only likely to confuse people.